File: java/src/programmers/가장큰수.java
package programmers;

import java.util.Arrays;
import java.util.stream.Collectors;

public class 가장큰수 {

	public String solution(int[] numbers) {
		var result = Arrays.stream(numbers)
			.mapToObj(String::valueOf)
			.sorted((a, b) -> (b + a).compareTo(a + b))
			.collect(Collectors.joining());

		return result.startsWith("0") ? "0" : result;
	}
}

File: python/programmers/가장큰수.py
from functools import cmp_to_key


def compare(a, b):
    if a + b > b + a:
        return -1
    elif a + b < b + a:
        return 1
    else:
        return 0


def solution(numbers):
    ls = list(map(str, numbers))
    numbers_sorted = sorted(ls, key=cmp_to_key(compare))
    answer = "".join(numbers_sorted)
    return "0" if answer[0] == "0" else answer
